### DISCUSSION TITLE
[Community Task #1] Update WTF Cairo 1 tutorials

### DISCUSSION THREAD

**Discussion by AmazingAng** [WTFAcademy/WTF-Starknet#54] [General]
## Task: Update WTF Cairo 1 tutorial

Let's create a minimalist Cairo 1 tutorial.

### Requirements:
- Have basic knowledge about Starknet.
- Have played with [starklings-cairo1](https://github.com/shramee/starklings-cairo1) or have written cairo1 contract in the past.
- Have some experience in rust
### Bounty
$60 per chapter ($960 in total).

### Outline
1. Hello Cairo 1 @AmazingAng
2. Felt @hasselalcala
3. Primitive Types @hasselalcala
4. Cairo programs vs Starknet contract @hasselalcala
5. Contract directives and Imports @hasselalcala
6. Functions: Externals vs View vs Internals vs Constructors @robertkp
7. Return state @Darlington02 
8. Cairo data types - (Arrays, structs, mappings, constants, tuples, Enums) @robertkp
9. Pattern matching @robertkp
10. Control flow @Darlington02 
11. Events [liuyepiaoxiang](https://github.com/liuyepiaoxiang) completed by 0xAA
12. Move - Ownership and Borrowing @shramee
13. Traits and implementations @shramee
14. Option [BoynChan](https://github.com/BoynChan) completed by 0xAA
15. assert [liuyepiaoxiang](https://github.com/liuyepiaoxiang) completed by 0xAA
16. Generics @sugarforever 

### How to apply
Reply below with the chapters you want to complete and a brief intro of yourself.

### Resources
1. WTF Cairo (v0) https://github.com/WTFAcademy/WTF-Cairo
2. WTF Solidity: https://github.com/AmazingAng/WTF-Solidity/blob/main/Languages/en/README.md
3. Starklings Cairo1: https://github.com/shramee/starklings-cairo1

**Comment by liuyepiaoxiang** [comment]
11 Events
15 assert

**Comment by hasselalcala** [comment]
I will be workin in chapters 2, 3, 4 and 5

**Comment by BoynChan** [comment]
I will be working on 14. Option

**Comment by sugarforever** [comment]
I will take 16. Generics.
